spinn_front_end_common/utility_models/data_speed_up_packet_gatherer_machine_vertex.py

```diff
@@ -117,31 +176,101 @@ class DataSpeedUpPacketGatherMachineVertex(
     # the amount of bytes the data length will take up
     LENGTH_OF_DATA_SIZE = 4
 
+    # point where sdp beats data speed up due to overheads
     THRESHOLD_WHERE_SDP_BETTER_THAN_DATA_EXTRACTOR_IN_BYTES = 40000
-
-    def __init__(
-            self, x, y, ip_address, report_default_directory,
-            write_data_speed_up_report, constraints=None):
-        super(DataSpeedUpPacketGatherMachineVertex, self).__init__(
+    THRESHOLD_WHERE_SDP_BETTER_THAN_DATA_INPUT_IN_BYTES = 300
+
+    # offset where data in starts on first command (
+    # command, base_address, x&y, max_seq_number)
+    OFFSET_AFTER_COMMAND_AND_ADDRESS_IN_BYTES = 16
+
+    # offset where data starts after a command id and seq number
+    OFFSET_AFTER_COMMAND_AND_SEQUENCE_IN_BYTES = 8
+
+    # size fo data to store when first packet with command and address
+    DATA_IN_FULL_PACKET_WITH_ADDRESS_NUM = \
+        DATA_PER_FULL_PACKET - (OFFSET_AFTER_COMMAND_AND_ADDRESS_IN_BYTES /
+                                WORD_TO_BYTE_CONVERTER)
+
+    # size for data in to store when not first packet
+    DATA_IN_FULL_PACKET_WITH_NO_ADDRESS_NUM = \
+        DATA_PER_FULL_PACKET - (OFFSET_AFTER_COMMAND_AND_SEQUENCE_IN_BYTES /
+                                WORD_TO_BYTE_CONVERTER)
+
+    # SDRAM requirement for storing missing SDP packets seq nums
+    SDRAM_FOR_MISSING_SDP_SEQ_NUMS = int(math.ceil(
+        (120 * 1024 * 1024) /
+        (DATA_PER_FULL_PACKET_WITH_SEQUENCE_NUM * WORD_TO_BYTE_CONVERTER)))
+
+    # size of data in key space
+    # x, y, key (all ints) for possible 48 chips,
+    SIZE_DATA_IN_CHIP_TO_KEY_SPACE = (3 * 4 * 48) + 4
+
+    # end flag for missing seq nums
+    MISSING_SEQ_NUMS_END_FLAG = 0xFFFFFFFF
+
+    # precompiled structures
+    _ONE_WORD = struct.Struct("<I")
+    _TWO_WORDS = struct.Struct("<II")
+    _THREE_WORDS = struct.Struct("<III")
+    _FOUR_WORDS = struct.Struct("<IIII")
+    _ADDRESS_PACKET_BYTE_FORMAT = struct.Struct(
+        "<{}B".format(DATA_IN_FULL_PACKET_WITH_ADDRESS_NUM *
+                      WORD_TO_BYTE_CONVERTER))
+
+    def __init__(self, x, y, ip_address, extra_monitors_by_chip, transceiver,
+                 report_default_directory, write_data_speed_up_report,
+                 write_data_in_report, constraints=None):
+        MachineVertex.__init__(
+            self,
             label="mc_data_speed_up_packet_gatherer_on_{}_{}".format(x, y),
             constraints=constraints)
+        AbstractHasAssociatedBinary.__init__(self)
+        AbstractProvidesLocalProvenanceData.__init__(self)
```

Comment:
Yet more `__init__` calling for cargo cult purposes.

Reply:
see above

Reply:
You'd call constructors on interfaces if you could? Stop it.